src/components/RoomNameInput.js

```diff
@@ -83,44 +89,46 @@ class RoomNameInput extends Component {
     }
 
     /**
-     * Modifies the room name to follow our conventions:
-     * - Max length 80 characters
-     * - Cannot not include space or special characters, and we automatically apply an underscore for spaces
-     * - Must be lowercase
-     * Also checks to see if this room name already exists, and displays an error message if so.
-     * @param {String} roomName
-     *
+     * @param {String} modifiedRoomName
      */
-    checkAndModifyRoomName(roomName) {
-        const modifiedRoomNameWithoutHash = roomName.substring(1)
-            .replace(/ /g, '_')
-            .replace(/[^a-zA-Z\d_]/g, '')
-            .substring(0, CONST.REPORT.MAX_ROOM_NAME_LENGTH)
-            .toLowerCase();
-        const finalRoomName = `#${modifiedRoomNameWithoutHash}`;
-
+    validateRoomName(modifiedRoomName) {
```

Comment:
Validation should happen in whatever parent component is using this. If we need to re-use it we should turn it into a utility method not have validation happen inside the input. Looking at this component again and the design is not compatible with our new forms and must change.

Reply:
@marcaaron Sure I will work on it. I have mentioned the same thing on [this comment](https://github.com/Expensify/App/pull/7098#issuecomment-1018188706) that validation is on child input component. That should be transferred to parent form component. So, what should we do Marc? Should we use validation on parent component only on where ever the `RoomNameInput` component are implemented? or make utility function on `libs` folder?

Reply:
There is the same set of validations for room names everywhere. So you can move these to a utility and then if you need to skip a few validations then that can be achieved vai params.

Reply:
Yes 

- remove the validation from the input and move it to `ValidationUtils.js`
- use this to validate in the parent component for each case
- See various methods used to validate input for reference...

https://github.com/Expensify/App/blob/ed588095ee477c489b5278f1bd251476b5281261/src/libs/ValidationUtils.js#L324-L346

Reply:
>Should we use validation on parent component only on where ever the RoomNameInput component are implemented

Yes

Reply:
Could you please do your changes and then we can review those?

Reply:
@ABee-Tech I'm not sure that name makes sense as there is another error related to the `policyID` and that validation should be separate from something called `isValidRoomName()`.

Reply:
Okay that makes sense. Thanks. I will apply the changes and will inform.

Reply:
I have done changes. Please review them.